compiler/rustc_hir_analysis/src/collect/predicates_of.rs

```diff
@@ -395,6 +395,8 @@ pub(super) fn trait_explicit_predicates_and_bounds(
     gather_explicit_predicates_of(tcx, def_id.to_def_id())
 }
 
+// FIXME(fmease): Needs investigation: Do & can we need to handle inherent associated types more
+// ‘precisely’ here? E.g. to break cycles.
```

Comment:
Would be nice to link to an issue or known-bug ui test that explains when we get cycles here.